q/compare.go

```diff
@@ -48,6 +48,18 @@ func compare(a, b interface{}, tok token.Token) bool {
 
 			return constant.Compare(constant.MakeFloat64(vala.Float()), tok, constant.MakeFloat64(bla))
 		}
+	case ak == reflect.Struct:
```

Comment:
@sbani i'm sorry but it is not exactly what i suggested. What's missing is that we must test the behavior with two types with the same fields.

```go
type A struct {
  Name string
  Age int
}

type B struct {
  Name string
  Age int
}
...compare(&A{Name:"john", Age:18), &A{Name:"john", Age:18}, token.EQ)... // must be true
...compare(&A{Name:"john", Age:18), &B{Name:"john", Age:18}, token.EQ)... // must be true
n := time.Now()
...compare(n, n, token.EQ)... // must be true
```